maze_transformer/generation/lattice_maze.py

```diff
@@ -306,17 +364,426 @@ def from_tokens(cls, tokens: list[str]) -> "LatticeMaze":
 
         return cls.from_adj_list(adj_list)
 
+    # ============================================================
+    # to and from pixels
+    # ============================================================
+    def _as_pixels_bw(self) -> BinaryPixelGrid:
+        assert self.lattice_dim == 2, "only 2D mazes are supported"
+        # Create an empty pixel grid with walls
+        pixel_grid: Int[np.ndarray, "x y"] = np.full(
+            (self.grid_shape[0] * 2 + 1, self.grid_shape[1] * 2 + 1),
+            False,
+            dtype=np.bool_,
+        )
+
+        # Set white nodes
+        pixel_grid[1::2, 1::2] = True
+
+        # Set white connections (downward)
+        for i, row in enumerate(self.connection_list[0]):
+            for j, connected in enumerate(row):
+                if connected:
+                    pixel_grid[i * 2 + 2, j * 2 + 1] = True
+
+        # Set white connections (rightward)
+        for i, row in enumerate(self.connection_list[1]):
+            for j, connected in enumerate(row):
+                if connected:
+                    pixel_grid[i * 2 + 1, j * 2 + 2] = True
+
+        return pixel_grid
+
+    def as_pixels(
+        self,
+        show_endpoints: bool = True,
+        show_solution: bool = True,
+    ) -> PixelGrid:
+        if show_solution and not show_endpoints:
+            raise ValueError("show_solution=True requires show_endpoints=True")
+        # convert original bool pixel grid to RGB
+        pixel_grid_bw: BinaryPixelGrid = self._as_pixels_bw()
+        pixel_grid: PixelGrid = np.full(
+            (*pixel_grid_bw.shape, 3), PixelColors.WALL, dtype=np.uint8
+        )
+        pixel_grid[pixel_grid_bw == True] = PixelColors.OPEN
+
+        if self.__class__ == LatticeMaze:
+            return pixel_grid
+
+        # set endpoints for TargetedLatticeMaze
+        if self.__class__ == TargetedLatticeMaze:
+            if show_endpoints:
+                pixel_grid[
+                    self.start_pos[0] * 2 + 1, self.start_pos[1] * 2 + 1
+                ] = PixelColors.START
+                pixel_grid[
+                    self.end_pos[0] * 2 + 1, self.end_pos[1] * 2 + 1
+                ] = PixelColors.END
+            return pixel_grid
+
+        # set solution
+        if show_solution:
+            for coord in self.solution:
+                pixel_grid[coord[0] * 2 + 1, coord[1] * 2 + 1] = PixelColors.PATH
+
+            # set pixels between coords
+            for index, coord in enumerate(self.solution[:-1]):
+                next_coord = self.solution[index + 1]
+                # check they are adjacent using norm
+                assert (
+                    np.linalg.norm(np.array(coord) - np.array(next_coord)) == 1
```

Comment:
Seems valuable to to have a method that takes 2 coords and returns a bool indicating if they're adjacent or not. Maybe this would even make sense as an `is_adjacent` method on a `Coord` class (might be out of scope)